Milestone4/OfflineStudentManagementSystem/Model/factoryComponents/creatorFactory.ts

```diff
@@ -0,0 +1,18 @@
+import { Teacher } from './teacher';
+import { Student } from './student';
+
+// type Constructor<T> = { new(): T }//inline type to capture this
+
+export class CreatorFactory {
+  static createUser<T>(user: T) {
+    const userName = <string>user;
```

Comment:
Why are we casting here?

Reply:
To accept anykind of datatype and convert it to string but i'm passing here as an argument as string so it is not necessary to keep it as generic it has been resolved